main.go

```diff
@@ -368,3 +372,22 @@ func runProfiler(addr string) {
 		setupLog.Error(err, "problem running profiler server")
 	}
 }
+
+func isCRDLoaded(mgr ctrlmgr.Manager, gvr schema.GroupVersionResource) (bool, error) {
```

Comment:
Maybe isCRDDeployed would be better

Reply:
Do we have e2e coverage that this new check works?

Reply:
This is found very fast, because the controller won't get added and e2e tests would fail. (That's also how I recognized it).

Do you think that's already sufficient?

Reply:
I was mostly wondering if we have e2e tests for both cases (now especially supervisor :D).

Reply:
I think we should be good. I guess we have at least test coverage for the first case and if it works there it should also work for supervisor

Reply:
Both cases are tested:

For supervisor: via `make test-integration`.

For non-supervisor: via `make e2e`.